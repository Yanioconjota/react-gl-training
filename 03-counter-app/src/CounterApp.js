import { useState } from "react";
import {PropTypes as propTypes} from "prop-types";
import { Row, Col, Button } from "react-bootstrap";

const CounterApp = ({ value, title, msg, showButtons, img }) => {

  const [counter, setCounter] = useState(value);
  console.log(counter);

  const handleAdd = () => {
    //setCounter( counter + 1 );
    setCounter(c => c + 1);
  }

  const handleSubstract = () => {
    (counter > 0) && setCounter(counter - 1);
  };

  const handleReset = () => {
    setCounter(value);
  };

  return (
    <>
      <Row>
        <Col md={6} xs={12}>
          <h1 className="text-info">{title}</h1>
          <p>
            {msg}
            {!showButtons && <span className="Code">{value}</span>}
            {showButtons && <span className="Code">{counter}</span>}
          </p>
          <div className="btn-group">
            {showButtons && (
              <Button onClick={handleAdd} variant="primary">
                +1
              </Button>
            )}
            {showButtons && (
              <Button onClick={handleReset} variant="info">
                Reset
              </Button>
            )}
            {showButtons && (
              <Button onClick={handleSubstract} variant="secondary">
                -1
              </Button>
            )}
          </div>
        </Col>
        {img && counter >= 5 && (
          <Col md={6} xs={12}>
            <p className="Code">Vamoh todav√≠a!</p>
            <img src={img} className="w-50" alt="Roque" />
          </Col>
        )}
      </Row>
    </>
  );
}

CounterApp.propTypes = {
  value: propTypes.number,
  title: propTypes.string,
  counter: propTypes.number,
  showButtons: propTypes.bool,
  img: propTypes.string
};

CounterApp.defaultProps = {
  title: "Counter App",
  msg: "",
  showButtons: true
};

export default CounterApp;
