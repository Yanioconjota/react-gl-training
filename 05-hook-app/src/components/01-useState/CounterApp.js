import React, { useState } from 'react';
import { Button } from 'react-bootstrap';
import { Row, Col } from "react-bootstrap";

const CounterApp = () => {

  const [state, setState] = useState({
    counter1: 10,
    counter2: 20,
    counter3: 30,
    counter4: 40
  });

  const {counter1, counter2, counter3, counter4} = state;

  //console.log(counter);

  return (
    <>
      <Row>
        <Col xs={6}>
          <h1>Counter {counter1}</h1>
          <h1>Counter {counter2}</h1>
        </Col>
        <Col xs={6}>
          <h1>Counter {counter3}</h1>
          <h1>Counter {counter4}</h1>
        </Col>
      </Row>
      <hr />
      <Button
        variant="primary"
        onClick={() => {
          setState({
            ...state,
            counter1: counter1 + 1,
            counter4: counter4 + 10
          });
        }}
      >
        +1
      </Button>
    </>
  );
};

export default CounterApp;