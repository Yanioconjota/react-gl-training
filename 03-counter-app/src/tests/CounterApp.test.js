import "@testing-library/jest-dom";
import { mount, shallow } from "enzyme";
import CounterApp from "../CounterApp";

describe('Pruebas en <CounterApp />', () => {
  //Utilizo mount en vez de shallow porque Button es un componente hijo de CounterApp que no puede ser encontrado usando shallow()
  let wrapper = mount(<CounterApp />);

  beforeEach(() => {
    wrapper = mount(<CounterApp showButtons={true} value={10} />);
  });
  test('debe mostrar <CounterApp /> correctamente ', () => {
    
    expect(wrapper).toMatchSnapshot();
  });
  test('debe mostrar el valor por defecto', () => {
    const counterText = wrapper.find('p > span.Code').text();
    expect(counterText).toBe('10');
  });
  test('debe de incrementar con el botón de +1', () => {
    wrapper.find(".btn-primary").simulate('click');
    const counterText = wrapper.find("p > span.Code").text();
    expect(counterText).toBe('11');
  });

  test("debe de incrementar con el botón de -1", () => {
    wrapper.find(".btn-secondary").simulate("click");
    const counterText = wrapper.find("p > span.Code").text();
    expect(counterText).toBe("9");
  });

  test("debe de reiniciar el valor con el botón de reset", () => {
    const wrapper = mount(<CounterApp showButtons={true} value={10} />);
    wrapper.find(".btn-secondary").simulate("click");
    wrapper.find(".btn-info").simulate("click");
    const counterText = wrapper.find("p > span.Code").text();
    expect(counterText).toBe("10");
  });
});
