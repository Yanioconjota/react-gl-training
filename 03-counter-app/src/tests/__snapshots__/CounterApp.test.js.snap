// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en <CounterApp /> debe mostrar <CounterApp /> correctamente  1`] = `
<div
  className="row"
>
  <div
    className="col-md-6 col-12"
  >
    <h1
      className="text-info"
    >
      Counter App
    </h1>
    <p>
      <span
        className="Code"
      >
        10
      </span>
    </p>
    <div
      className="btn-group"
    >
      <button
        className="btn btn-primary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        +1
      </button>
      <button
        className="btn btn-info"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        Reset
      </button>
      <button
        className="btn btn-secondary"
        disabled={false}
        onClick={[Function]}
        type="button"
      >
        -1
      </button>
    </div>
  </div>
</div>
`;
